### PR TITLE
boot/seal_test.go: split some tests with overlord/fdestate/backend

### PR DESCRIPTION
**Pull request by valentindavid**
Part 3 of #14282

### CODE REVIEW

File: boot/seal_test.go
Comment (pedronis) [inline boot/seal_test.go:1]: should we move 

func (s *sealSuite) TestResealKeyToModeenvWithFdeHookCalled(c *C) {
func (s *sealSuite) TestResealKeyToModeenvWithFdeHookVerySad(c *C) {

even if they do very little?

Comment (valentindavid) [reply boot/seal_test.go:1]: I suppose we will write test as soon as we implement it. It really does nothing at all now.

File: overlord/fdestate/backend/reseal_test.go [hunk 1]
```diff
@@ -275,3 +328,1185 @@ func (s *resealTestSuite) TestTPMResealHappy(c *C) {
 	c.Assert(cnt, Equals, 1)
 	c.Check(recoveryPbc, DeepEquals, boot.ToPredictableBootChains(removeKernelBootFiles(params.RecoveryBootChains)))
 }
+
+func (s *resealTestSuite) TestResealKeyToModeenvWithSystemFallback(c *C) {
```
[the remark this answers is not in the document]
Comment (valentindavid) [reply overlord/fdestate/backend/reseal_test.go]: I have renamed them.

**Review by pedronis — APPROVED** [review]
thank you

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks

**Pull request merged**